### PR TITLE
assert_dom strict parameter

### LINKED ISSUE #121: `assert_dom` should ignore whitespace just like `assert_dom_equal`
In one of my views I have this:

```ruby
<li><%= item.quantity %> &times; <%= item.product.title %></li>
```

My formatter automatically breaks this onto new lines like so:

```ruby
      <li><%= item.quantity %>
        &times;
        <%= item.product.title %></li>
```

Because of this, running a test that contains this assertion fails:

```ruby
assert_select "ul li", "#{item.quantity} \u00D7 #{products(:one).title}"
```

This is due to the text I'm passing to `assert_select` having no newlines in it, but in the view code there _are_ newlines even though the browser actually drops those newlines.

So the test output is as follow:

```
Failure:
LineItemsControllerTest#test_should_create_line_item [test/controllers/line_items_controller_test.rb:27]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"1 × The Pragmatic Programmer"
+"1
+        ×
+        The Pragmatic Programmer"
.
Expected 0 to be >= 1.
```

My current fix is to either turn off my formatter for that file, or to update the assertion to use regex and replace the spaces with `\s+` like this:

```ruby
assert_select "ul li", /#{item.quantity}\s+\u00D7\s+#{products(:one).title}/
```

However, the test should really treat excess whitespace the same as the browser does by dropping it.

Sounds like the same thing was done for `assert_dom_equal` in #84.

### PR DESCRIPTION
**Pull request by jyeharry**
- **Add macos to supported platforms in Gemfile.lock**
- **Add failing test for assert_dom collapsing whitespace**
- **Add strict parameter to assert_dom**

Fixes #121

Adds a `:strict` key to the equality test hash for `assert_dom`, so that we can specify whether or not to ignore excess whitespace just as the browser does.

I prefer the solution in #123. This was created since the issue I created specifically mentioned creating a `:strict` parameter.

### CODE REVIEW

**Comment by flavorjones** [comment]
OK, master is green, so please rebase when you have a chance.

**Comment by jyeharry** [comment]
@flavorjones Rebased and tests and ci is passing

**Comment by jyeharry** [comment]
@flavorjones just following up on this
